### PR TITLE
SDK-1585: Integrator should check if the configuration index already exists

### PR DESCRIPTION
**Pull request by Kyrylo-Meretskyi**
Ticket: [SDK-1585](https://spryker.atlassian.net/browse/SDK-1585)

### CODE REVIEW

File: src/ManifestStrategy/ConfigureEnvManifestStrategy.php [hunk 1]
```diff
@@ -90,6 +90,33 @@ public function apply(array $manifest, string $moduleName, InputOutputInterface
         return true;
     }
 
+    /**
+     * @param string $target
+     * @param \SprykerSdk\Integrator\Dependency\Console\InputOutputInterface $inputOutput
+     *
+     * @return bool
+     */
+    protected function isTargetExist(string $target, InputOutputInterface $inputOutput): bool
```
Comment (dereuromark) [inline src/ManifestStrategy/ConfigureEnvManifestStrategy.php:99]: thats not a valid method name

targetExists() IMO

Comment (Kyrylo-Meretskyi) [reply src/ManifestStrategy/ConfigureEnvManifestStrategy.php]: Fixed

File: src/ManifestStrategy/ConfigureEnvManifestStrategy.php [hunk 2]
```diff
@@ -90,6 +90,33 @@ public function apply(array $manifest, string $moduleName, InputOutputInterface
...
+     * @return bool
+     */
+    protected function targetExists(string $target, InputOutputInterface $inputOutput): bool
+    {
+        $configFileName = $this->config->getConfigPath();
+        $configFileContent = file_get_contents($configFileName);
+
+        if ($configFileContent === false) {
+            $inputOutput->writeln(sprintf(
+                'Could not read from file `%s`. Please check file path or customize using `IntegratorConfig::getConfigPath()`.',
```
Comment (pavelmaksimov25) [inline src/ManifestStrategy/ConfigureEnvManifestStrategy.php:106]: `IntegratorConfig::getConfigPath()` doesn't look like something we can customize a string with.

**Review by pavelmaksimov25 — COMMENTED** [review]
It would be great to see some tests that cover this fix.

File: src/ManifestStrategy/ConfigureEnvManifestStrategy.php [hunk 3]
```diff
@@ -90,6 +90,33 @@ public function apply(array $manifest, string $moduleName, InputOutputInterface
...
+                $configFileName,
+            ), InputOutputInterface::DEBUG);
```
[the remark this answers is not in the document]
Comment (DmytroKlymanSpryker) [reply src/ManifestStrategy/ConfigureEnvManifestStrategy.php:108]: We already have this code on line 70 with file_exists(). Why do we need to copy/paste it to this method with file_get_contents() ?

[the remark this answers is not in the document]
Comment (Kyrylo-Meretskyi) [reply src/ManifestStrategy/ConfigureEnvManifestStrategy.php]: Fixed

**Comment by DmytroKlymanSpryker** [comment]
Please solve Code Styling problems:

```
FILE: ...egrator/tests/SprykerSdkTest/Integrator/IntegratorFacadeTest.php
----------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
 290 | ERROR | [x] Multi-line function calls must have a trailing
     |       |     comma after the last parameter.
     |       |     (SlevomatCodingStandard.Functions.RequireTrailingCommaInCall.MissingTrailingComma)
 328 | ERROR | [x] Multi-line function calls must have a trailing
     |       |     comma after the last parameter.
     |       |     (SlevomatCodingStandard.Functions.RequireTrailingCommaInCall.MissingTrailingComma)
 509 | ERROR | [x] Multi-line function calls must have a trailing
     |       |     comma after the last parameter.
     |       |     (SlevomatCodingStandard.Functions.RequireTrailingCommaInCall.MissingTrailingComma)
 510 | ERROR | [x] Multi-line function calls must have a trailing
     |       |     comma after the last parameter.
     |       |     (SlevomatCodingStandard.Functions.RequireTrailingCommaInCall.MissingTrailingComma)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

File: src/ManifestStrategy/ConfigureEnvManifestStrategy.php [hunk 4]
```diff
@@ -90,6 +90,19 @@ public function apply(array $manifest, string $moduleName, InputOutputInterface
...
 
+    /**
+     * @param string $target
+     *
+     * @return bool
+     */
+    protected function targetExists(string $target): bool
+    {
+        $configFileName = $this->config->getConfigPath();
+        $configFileContent = (string)file_get_contents($configFileName);
```
Comment (DmytroKlymanSpryker) [inline src/ManifestStrategy/ConfigureEnvManifestStrategy.php:101]: `$configFileName` variable is used only once here, so it is not really needed

Comment (Kyrylo-Meretskyi) [reply src/ManifestStrategy/ConfigureEnvManifestStrategy.php]: Missed this after fixes
Fixed

**Review by pavelmaksimov25 — APPROVED** [review]

**Review by DmytroKlymanSpryker — APPROVED** [review]

**Pull request merged**